hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestIncrementalReadWithFullTableScan.scala

```diff
@@ -70,9 +70,11 @@ class TestIncrementalReadWithFullTableScan extends HoodieSparkClientTestBase {
       HoodieWriteConfig.TBL_NAME.key -> "hoodie_test",
       HoodieMetadataConfig.COMPACT_NUM_DELTA_COMMITS.key -> "1"
     )
+
+
     // Create 10 commits
     for (i <- 1 to 10) {
-      val records = recordsToStrings(dataGen.generateInserts("%05d".format(i), perBatchSize)).asScala.toList
+      val records = recordsToStrings(dataGen.generateInserts(System.currentTimeMillis(), perBatchSize)).asScala.toList
```

Comment:
Have you added any new test cases that cover the scenarios where the completion time ordering is different than instant time ordering and make sure the incremental queries return the correct results, for both incremental and fallback full scan?

Reply:
There are some tests in `TestHoodieIncrSource` that has instant times don't match completion time to test hollow commits handling that can be used to test this feature. 

I've filed a jira to further improve the test coverage: https://issues.apache.org/jira/browse/HUDI-8356